Worktodo.cpp

```diff
@@ -98,12 +102,12 @@ std::optional<Task> Worktodo::getTask(Args &args) {
  again:
   // Try to get a task from the local worktodo.txt
   if (optional<Task> task = firstGoodTask(worktodoTxt)) {
-    if (task->kind == Task::PRP && task->wantsPm1) {
+    if ((task->kind == Task::LL and !task->wantsPm1) or (task->kind == Task::PRP and task->wantsPm1)) {
```

Comment:
I don't understand the logic here: if the task is LL and !wantsPm1, then you do P-1? and flip the "wants" flag to true in than case below?

Reply:
The `wantsPm1` value is read directly from the assignment line in the worktodo file. For PRP assignments, this is the `tests_saved` (PRP tests saved if factor is found), so a truthy (greater than zero) value means P-1 is needed. However, for LL assignments, this is the `has_been_pminus1ed` (has P-1 been done), so a falsy (zero) value means P-1 is needed, the opposite of PRP assignments. That is what this logic was trying to express, without changing your variable names, as I was trying to keep the changes to a minimum. Let me know if you would like me to make any changes...

Reply:
thanks for the explanation. The name "wantsPm1" ain't good, but I understand the motivation of keeping the change small.